parachain/src/primitives.rs

```diff
@@ -122,10 +122,14 @@ impl From<i32> for Id {
 }
 
 const USER_INDEX_START: u32 = 1000;
+const PUBLIC_INDEX_START: u32 = 2000;
```

Comment:
what's the difference between user index & public index?
does this means 1000 - 2000 are reserved for common good chain only? that can only be registered with root?

Reply:
Correct. System chains (0..=999) can have their transactions executed with root. Anything 1000 and above are user-level and cannot.

Within the user-level ID space, we begin the public IDs at 2000 to prevent clashing with IDs used for common-good chains.